src/room_member.cpp

```diff
@@ -170,6 +171,7 @@ static void SendJoinRequest(RakNet::RakPeerInterface* peer, const std::string& n
 void RoomMember::ReceiveLoop() {
     // Receive packets while the connection is open
     while (IsConnected()) {
+        std::this_thread::sleep_for(sleep_time);
         std::lock_guard<std::mutex> lock(network_mutex);
```

Comment:
As far as i know it loks on lock_guard at creating until the end of it's block. That is what i intended. I don't want to lock it during the sleep